### DISCUSSION TITLE
PCManFM-Qt segfaults when accessing a specific folder

### DISCUSSION THREAD

**Discussion by bbbhltz** [lxqt/lxqt#2506] [Q&A]
Hello all, 

Quick question. I have LXQt 1.4.0 running on Alpine Edge. I noticed today that there is a folder that causes PCManFM-Qt to segfault and crash.

How would I go about tracking down what is causing this?

The folder in question is `~/Projects/simpleblog`.

If I use the location bar I can get to the subfolders of that folder no problem. But, I cannot view that folder at all.

I just don't want to make an issue that seems like it could be a me problem.

**Comment by tsujan** [comment]
If you have `systemd`, check `/var/lib/systemd/coredump` (if in your distro `coredump` isn't installed by default, install and enable it — consult your distro doc — but in Arch and its derivatives it's installed and enabled). If there's a recent item for APP in it (APP's name may be `pcmanfm-qt` or `libfm-qt`), use this command in a terminal:

```
coredumpctl gdb APP
where
```

and attach what comes after `where`. Finally, type `q` in the terminal.

Also describe what happens as clearly as possible, in a step-by-step way.

If there's no recent item in `/var/lib/systemd/coredump` matching the time when the problem happened (check the modification dates), then probably it wasn't about a crash.

**Comment by bbbhltz** [comment]
Ok, thanks! I'll take a look at the docs. Alpine doesn't use systemd but I'll see what messages I can recover from gdb.

As for steps:

1. Click on pcmanfm-qt
2. Navigate from home to Projects
3. Navigate to simpleblog folder
4. Crash (wallpaper flashes)

**Reply by tsujan** [reply]
Here is an alternative method that I think should work everywhere:

1. Stop pcmanfm-qt's process completely.
2. In a terminal, type `gdb pcmanfm-qt`.
3. Type `run` at the `(gdb)`.
4. When the crash happens, type `where` and post the results.

**Reply by bbbhltz** [reply]
I didn't need to go that far. It was a "me problem".

I noticed a `.json` file in that directory that served no purpose so I deleted it and now everything works. I will mark your reply as the answer through, because that probably would have told me what as up.

Thanks for the quick replies!

**Reply by tsujan** [reply]
> It was a "me problem".

Are you sure? A crash report with a backtrace is **always** a good report. In rare cases, the backtrace (what comes after `where`) may show that the cause of the crash is outside the app, but in general, such reports are crucial to the improvement of an app.

**Reply by tsujan** [reply]
> I noticed a .json file in that directory that served no purpose

I mean that, if a crash happens with that `.json` file back into that directory, your report with a backtrace would be really appreciated.

**Reply by bbbhltz** [reply]
yeah you're right... any .json makes it crash

I created an empty `.json` in my home and gave it a go

<details>
  <summary>gdb</summary>

```bash
(gdb) run
Starting program: /usr/bin/pcmanfm-qt 
[New LWP 16181]
[New LWP 16182]
[New LWP 16183]
[New LWP 16184]
[New LWP 16185]
isPrimaryInstance
[New LWP 16187]
[LWP 16187 exited]
[New LWP 16188]
[New LWP 16189]
[New LWP 16190]

Thread 9 "Fm::JobThread" received signal SIGSEGV, Segmentation fault.
[Switching to LWP 16189]
get_meta (p=p@entry=0x7ffff3768c10 "\354%\225\363\377\177")
    at src/malloc/mallocng/meta.h:141
warning: 141    src/malloc/mallocng/meta.h: No such file or directory
(gdb) where
#0  get_meta (p=p@entry=0x7ffff3768c10 "\354%\225\363\377\177")
    at src/malloc/mallocng/meta.h:141
#1  0x00007ffff7f80bbe in __libc_free (p=0x7ffff3768c10)
    at src/malloc/mallocng/free.c:105
#2  0x00007ffff6b6917b in ??? () at /usr/lib/libgio-2.0.so.0
#3  0x00007ffff6af7214 in g_content_type_is_a () at /usr/lib/libgio-2.0.so.0
#4  0x00007ffff6af7836 in g_content_type_can_be_executable ()
    at /usr/lib/libgio-2.0.so.0
#5  0x00007ffff7df5ad9 in Fm::MimeType::MimeType(char const*) ()
    at /usr/lib/libfm-qt.so.14
#6  0x00007ffff7df5d47 in Fm::MimeType::fromName(char const*) ()
    at /usr/lib/libfm-qt.so.14
#7  0x00007ffff7df6189 in Fm::MimeType::guessFromFileName(char const*) ()
    at /usr/lib/libfm-qt.so.14
#8  0x00007ffff7df835d in Fm::FileInfo::setFromGFileInfo(Fm::GObjectPtr<_GFileInfo> const&, Fm::FilePath const&, Fm::FilePath const&) () at /usr/lib/libfm-qt.so.14     
#9  0x00007ffff7e05be9 in Fm::DirListJob::exec() () at /usr/lib/libfm-qt.so.14
#10 0x00007ffff7e01aca in Fm::Job::run() () at /usr/lib/libfm-qt.so.14
#11 0x00007ffff6cac9db in ??? () at /usr/lib/libQt5Core.so.5
```
</details>

**Reply by tsujan** [reply]
Thank you very much! I'll investigate it soon.

**Comment by tsujan** [comment]
The crash isn't reproducible here: an empty `.json` file is like other empty files and doesn't cause a problem. I also checked our code and found nothing wrong in it.

The segfault seems to happen in GLib (in `g_content_type_is_a`) due to some sort of mimetype issue. The root of the problem may be in GLib → `gio/xdgmime/xdgmime.c`.

Some questions:

1. Does it happen with other kinds of empty files too?
2. Does it happen after you update your mimetype database (with `update-mime-database /usr/share/mime` — please consult your distro doc before doing it!).
3. Does `update-mime-database` show any error message in terminal?

**Comment by bbbhltz** [comment]
It was an Alpine issue.

Today's upgrade took care of it. I didn't make a note of what packages upgraded but glib what the only library.

**Reply by tsujan** [reply]
> Today's upgrade took care of it.

Good to know that :)
